external/fv3fit/fv3fit/keras/_models/models.py

```diff
@@ -216,9 +216,9 @@ def fit(
         fit_kwargs = _fill_default(
             fit_kwargs, use_last_batch_to_validate, "use_last_batch_to_validate", False
         )
-
-        Xy = _XyArraySequence(self.X_packer, self.y_packer, batches)
-
+        random_state = np.random.RandomState(np.random.get_state()[1][0])
```

Comment:
Now that the shuffle method is inside `.fit` it doesn't have easy access to the `TrainingConfig.random_seed`. My solution here was to assume the random state at this point in time will be reproducible if the starting random state that is set at the beginning of `train.py` is the same. LMK if that is not the case. The reproducibility tests seem ok.

Reply:
> assume the random state at this point in time will be reproducible if the starting random state that is set at the beginning of train.py is the same.

Looks like this is not true, since the reproducibility regtests that passed locally failed on CircleCI. I think the random state used here needs to also be directly set with the same seed to keep the same behavior. I think we could remove the `TrainingConfig.random_seed` and instead make it part of the models' `Hyperparameters` so that it's available to `.fit` if it needs to set a random state, and call `fv3fit.set_random_seed(random_seed)` in the model training function instead of `train.py`. Let me know if you see any issues with that.

Reply:
Looks like you've reverted this now.